File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min = float('inf')
        maxProfit = 0
        for i in prices:
            if i < min:
                min = i
            maxProfit = max(maxProfit, i - min)
        return (maxProfit)

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    minprice := prices[0]
    maxprofit := 0
    for _, price := range prices {
        if price < minprice {
            minprice = price
        } else if price-minprice > maxprofit {
            maxprofit = price - minprice
        }
    }
    return maxprofit
}


